.github/workflows/codspeed.yml

```
@@ -0,0 +1,53 @@
+name: codspeed
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  # `workflow_dispatch` allows CodSpeed to trigger backtest
+  # performance analysis in order to generate initial data.
+  workflow_dispatch:
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+
+      - uses: actions/cache@v2
+        id: cache-py
+        name: cache python
+        with:
+          path: ${{ env.pythonLocation }}
+          key: >
+            py
+            ${{ env.pythonLocation }}
+            ${{ hashFiles('tests/requirements.txt') }}
+            ${{ hashFiles('pyproject.toml') }}
+
+      - run: pip install -r tests/requirements.txt
+        if: steps.cache-py.outputs.cache-hit != 'true'
+
+      - run: rm rust-toolchain
+      - name: install rust nightly
+        uses: dtolnay/rust-toolchain@nightly
```

```suggestion
      - name: install rust stable
        uses: dtolnay/rust-toolchain@stable
```
Probably more correct to test on stable, we should build binaries on stable too (I'll check that).

The only thing that needs nightly is no-coverage tags.